dotnet/src/SemanticKernel/Planning/FunctionFlowRunner.cs

```diff
@@ -154,12 +168,56 @@ private async Task<string> ProcessNodeListAsync(XmlNodeList nodeList, string fun
                     continue;
                 }
 
+                // Ignoring first ELSE after the If that was already processed
+                if (ignoreElse && o2.Name.StartsWith(ConditionElseTag, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    context.Log.LogTrace("{0}: ignoring ELSE tag node", parentNodeName);
+                    ignoreElse = false;
+                    continue;
+                }
+
+                if (processFunctions && o2.Name.StartsWith(ConditionIfTag, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    context.Log.LogTrace("{0}: found IF tag node", parentNodeName);
+                    // Includes IF + ELSE statement
+                    var ifFullContent = o2.OuterXml;
+
+                    //Go for the next node to see if it's an else
+                    if (this.CheckIfNextNodeIsElseAndGetItsContents(o2, out var elseContents))
+                    {
+                        ifFullContent += elseContents;
+
+                        // Ignore the next else sibling tag to avoid adding recursively it to the next plan iteration.
+                        ignoreElse = true;
+                    }
+
+                    var functionVariables = new ContextVariables();
+                    functionVariables.Update(context.Variables);
```

Comment:
Line 196 will just overwrite the behavior of line 195. Was clone intended?

Reply:
Nice, didn't know we had the clone in SKContext. Will update to it